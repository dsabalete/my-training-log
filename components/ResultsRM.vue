<template>
  <div v-if="rm1 != 0" class="result">
    <h4 v-for="(coef, index) in coeficients" :key="index">
      {{ index + 1 }}RM: {{ (coef * rm1) | twoDecimals }}
    </h4>
  </div>
</template>

<script>
export default {
  filters: {
    twoDecimals(value) {
      return value.toFixed(0) + ' Kg'
    }
  },
  props: {
    rm1: {
      type: Number,
      required: true
    }
  },
  data() {
    return {
      coeficients: [1, 0.95, 0.93, 0.9, 0.87, 0.85, 0.83, 0.8, 0.77, 0.75]
    }
  }
}
</script>

<style scoped>
.result {
  margin: 30px;
}
</style>
