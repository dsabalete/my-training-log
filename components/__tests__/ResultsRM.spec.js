import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import ResultsRM from '@/components/ResultsRM.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

const store = new Vuex.Store({
  state: {
    weight: 50,
    reps: 5
  }
})

const factory = () => {
  return shallowMount(ResultsRM, {
    propsData: {
      rm1: 90
    },
    store,
    localVue
  })
}

describe('ResultsRM', () => {
  let wrapper

  beforeEach(() => {
    wrapper = factory()
  })

  it('should be a Vue Instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders properly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have 10 h4 elements', () => {
    expect(wrapper.findAll('h4').length).toBe(10)
  })
})
