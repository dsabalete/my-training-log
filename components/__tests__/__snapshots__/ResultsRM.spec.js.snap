// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsRM renders properly 1`] = `
"<div class=\\"result\\">
  <h4>
    1RM: 90 Kg
  </h4>
  <h4>
    2RM: 86 Kg
  </h4>
  <h4>
    3RM: 84 Kg
  </h4>
  <h4>
    4RM: 81 Kg
  </h4>
  <h4>
    5RM: 78 Kg
  </h4>
  <h4>
    6RM: 77 Kg
  </h4>
  <h4>
    7RM: 75 Kg
  </h4>
  <h4>
    8RM: 72 Kg
  </h4>
  <h4>
    9RM: 69 Kg
  </h4>
  <h4>
    10RM: 68 Kg
  </h4>
</div>"
`;
